File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n=nums.size();
        int ans=0;
        int result=nums[0];
        for(int i=0;i<n;i++){
            ans+=nums[i];
            result=max(ans,result);
            if(ans<0){
                ans=0;
            }
        }
        return result;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        summ=0
        ans=nums[0]
        for i in range(len(nums)):
            if summ<0:
                summ=0
            summ+=nums[i]
            ans=max(summ,ans)
        return ans